doc/source/guide/derivatives_guide.rst

```diff
@@ -52,25 +57,31 @@ Here :math:`f'(x)` is defined as the linear operator (if it exists) that satisfi
 
 This definition of the derivative is called the *Fréchet derivative*.
 
+The Gateaux derivative
+~~~~~~~~~~~~~~~~~~~~~~
+The concept of directional derivative can also be extended to Banach spaces, giving the *Gateaux* derivative. 
+The Gateaux derivative is more general than the Fréchet derivative, but is not a linear operator. An example of a function that is Gateaux differentiable but not Fréchet differentiable is the absolute value function.
+It not used in ODL since it can fail to be linear. 
```

Comment:
In a way, already the `NumericalDerivative` falls into the nonlinear derivative category, since we only pretend it's linear. And implicitly, the gradient operators of not-everywhere-smooth functionals like `L1Norm` can violate linearity, depending on the value that is substituted in the points where it's non-differentiable.
So I think this topic reaches farther than just this PR, it would also require additional documentation (like "op.derivative must be linear" in various places). Make a new issue, probably we want to discuss this in more detail.